app/destination/page.js

```diff
@@ -29,36 +28,59 @@ export const Destinations = () => {
   let numberOfPlanets = 0;
 
   const onAddOrRemovePlanet = (name, index) => {
-    // TASK - React 1 week 2
-    // Implement this function
-    // If you press the "ADD PLANET" the selected planet should display "SELECTED"
-    // And the counter should update, how many planets are selected (numberOfPlanets)
-    console.log(`You seleceted the following planet: ${name}, with the index of ${index}`);
-  }
+    const planetIndex = selectedPlanets.indexOf(name);
+
+    if (planetIndex > -1) {
+      const updatedPlanets = [...selectedPlanets];
+      updatedPlanets.splice(planetIndex, 1);
+      onAddPlanet(updatedPlanets);
+    } else {
+      onAddPlanet([...selectedPlanets, name]);
+    }
+
+    console.log(
+      `You seleceted the following planet: ${name}, with the index of ${index}`
+    );
 
+    // TASK - React 1 week 2
+    // Implement this function       ---- done
+    // If you press the "ADD PLANET" the selected planet should display "SELECTED"  ---done
+    // And the counter should update, how many planets are selected (numberOfPlanets)   ---done
+    //console.log(`You seleceted the following planet: ${name}, with the index of ${index}`);   ---done
+  };
+  numberOfPlanets = selectedPlanets.length;
   return (
     <div className="fullBGpicture">
       <main className="mainContent">
         <h1>Travel destinations</h1>
         <section className="card">
           <h2>Wishlist</h2>
-          {/* TASK - React 1 week 2 */}
-          {/* Display the number Of selected planets */}
-          {/* Display the "no planets" message if it is empty! */}
-          <p>No planets in wishlist :(</p>
-          <p>You have {numberOfPlanets} in your wishlist</p>
+          {numberOfPlanets === 0 ? (
+            <p>No planets in wishlist :</p>
+          ) : (
+            <p>You have {numberOfPlanets} planet(s) in your wishlist</p>
```

Comment:
Here you could use a template string to add plural to planet only when the numberOfPlanets is bigger then 1  🙂